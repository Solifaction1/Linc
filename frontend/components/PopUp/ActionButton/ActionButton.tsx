import React from 'react'

import { defaultProps, propsConfig } from './actionButton.config'
import styles from './actionButton.module.scss'

const ActionButton: React.FC<propsConfig> = (props: propsConfig) => {
	return (
		<button
			type="button"
			className={`
			${props.className} 
			${styles.actionButton} 
			${styles[props.buttonType || 'primary']} 
			${styles[props.size || 'md']}`}
			disabled={props.disabled}
			onClick={props.handleClick || ((e) => e.preventDefault())}
		>
			<div />
			<div />
		</button>
	)
}

ActionButton.defaultProps = defaultProps

export default ActionButton
