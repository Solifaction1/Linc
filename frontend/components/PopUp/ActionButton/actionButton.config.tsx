import React from "react";

type propsConfig = {
  buttonType?: "close" | string;
  size?: "sm" | "md" | "lg" | string;
  disabled?: boolean;
  className?: string;
  handleClick?: React.MouseEventHandler<HTMLButtonElement>;
};

const defaultProps = {
  buttonType: "primary",
  size: "md",
  disabled: false,
  className: "",
  handleClick: (): void => console.log("test"),
};

export type { propsConfig };
export { defaultProps };
