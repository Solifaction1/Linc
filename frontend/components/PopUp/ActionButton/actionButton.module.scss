@import "styles/vars";
@import "styles/utilities";

.actionButton {
  $button-padding: 1rem;
  position: relative;

  width: 3rem;
  height: 3rem;
  padding: $button-padding;
  border: none;

  background: $color-gray-tint-80;

  cursor: pointer;
  
  &:hover{
    background: $color-gray-tint-70;
  }
  
  &:active{
    background: $color-gray-tint-60;
  }

  &.sm{
    width: 2rem;
    height: 2rem;
  }
  
  &.md{
    width: 3rem;
    height: 3rem;
  }
  
  &.lg{
    width: 4rem;
    height: 4rem;
  }


  &[disabled],
  &[disabled]:hover,
  &[disabled]::before {
    opacity: .5;
    pointer-events: none;
  }


  &.close{
    div{
      content: "";
      position: absolute;
      top: 50%;
      left: 50%;

      width: .3rem;
      height: calc(100% - $button-padding);

      background: $color-gray-shade-80;

      transform: translate(-50%, -50%);
    
    
      &:nth-child(1){
        transform: translate(-50%, -50%) rotate(-45deg);
      }
    
      &:nth-child(2){
        transform: translate(-50%, -50%) rotate(45deg);

      }
    }
  }
}