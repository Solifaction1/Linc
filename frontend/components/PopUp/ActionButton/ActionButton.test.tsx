import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import ActionButton from './ActionButton'

describe('ActionButton', () => {
	it('renders an ActionButton', () => {
		const { container } = render(
			<ActionButton buttonType="close" size="lg" className="ActionButton" />
		)

		expect(container).toMatchSnapshot()

		const button = container.querySelector('button')
		expect(button).toBeVisible()
		expect(button?.getAttribute('class')).toContain('ActionButton')
		expect(button?.getAttribute('class')).toContain('actionButton')
		expect(button?.getAttribute('class')).toContain('close')
		expect(button?.getAttribute('class')).toContain('lg')

		expect(button?.getAttribute('type')).toContain('button')
	})
})
