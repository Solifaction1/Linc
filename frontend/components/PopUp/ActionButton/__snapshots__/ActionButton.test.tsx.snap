// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton renders an ActionButton 1`] = `
<div>
  <button
    class="
			ActionButton 
			actionButton 
			close 
			lg"
    type="button"
  >
    <div />
    <div />
  </button>
</div>
`;
